### PR TITLE
Improved logging for NodeProvisioner

### PR DESCRIPTION
**Pull request by jglick**
Have been having issues with `kubernetes` agent provisioning and found this helpful to keep track of what is going on.

### Proposed changelog entries

* None needed, this is only adding sub-`INFO` logging.

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]

**Review by daniel-beck — APPROVED** [review]

**Review by Vlatombe — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Will merge it into the next weekly if nobody is against

**Comment by oleg-nenashev** [comment]
Does it need backporting @Vlatombe @jglick ? If so, please create a JIRA ticket for this change

File: core/src/main/java/hudson/slaves/NodeProvisioner.java [hunk 1]
```diff
@@ -327,6 +331,9 @@ public void run() {
         } finally {
             provisioningLock.unlock();
         }
+        if (LOGGER.isLoggable(Level.FINER)) {
+            LOGGER.finer(() -> "ran update on " + label + " in " + (System.nanoTime() - start) / 1_000_000 + "ms");
```
Comment (jtnord) [inline core/src/main/java/hudson/slaves/NodeProvisioner.java]: I thought the whole point of the `MessageSupplier` was that it was not invoked unless the logging was enabled thus should be safe without the calling wrap?  (like you have above in [line182](https://github.com/jenkinsci/jenkins/pull/4081/files#diff-0558cec054efaecfd6bbee9cbbccdab4R182)  asking for my education as I guess there is a reason you did it.

**Review by jtnord — APPROVED** [review]

File: core/src/main/java/hudson/slaves/NodeProvisioner.java (hunk 1 above)
Comment (jglick) [reply core/src/main/java/hudson/slaves/NodeProvisioner.java]: Merely that my IDE hint to the effect that you should use either a message template or a lambda is not clever enough to notice that the statement is gated inside `isLoggable`. Happy to delete `() ->` if it is confusing you.

**Comment by jglick** [comment]
> Does it need backporting

I do not suppose so. The real change would be #4082.

**Pull request merged**